### ISSUE TITLE
Review and tidy API endpoint catalogue

### ISSUE THREAD

**Issue by nathanielknight** [cfe-lab/Kive#968]
The `KiveAPI` class has a [catalogue](https://github.com/cfe-lab/Kive/blob/master/api/kiveapi/kiveapi.py#L37-L59) that it uses to provide easy access to commonly used endpoints. This catalogue seems to make reference to deprecated URLs (e.g. `/api/pipelines`). To remedy this:

- [ ] The catalogue should be audited to see which URLs correspond to current endpoints.
- [ ] Deprecated URLs should be replaced with up-to-date ones.

It might be worth considering a test-case to check that all the URLs in the catalogue (or even just the URLs that don't have parameters) actually route successfully.

**Comment by nathanielknight** [comment]
For reference, as of the time of writing, these are the API endpoints that the Django Rest Framework API browser exposes:

```json
{
    "batches": "http://localhost:8080/api/batches/?page_size=25",
    "containerchoices": "http://localhost:8080/api/containerfamilies/?page_size=25",
    "containerfamilies": "http://localhost:8080/api/containerfamilies/?page_size=25",
    "containers": "http://localhost:8080/api/containers/?page_size=25",
    "containerapps": "http://localhost:8080/api/containerapps/?page_size=25",
    "containerargs": "http://localhost:8080/api/containerargs/?page_size=25",
    "containerruns": "http://localhost:8080/api/containerruns/?page_size=25",
    "containerlogs": "http://localhost:8080/api/containerlogs/?page_size=25",
    "datasets": "http://localhost:8080/api/datasets/?page_size=25",
    "externalfiledirectories": "http://localhost:8080/api/externalfiledirectories/?page_size=25",
    "users": "http://localhost:8080/api/users/?page_size=25"
}```

**Comment by donkirkby** [comment]
That's actually intended for internal use, like [this]:

    dataset = self.get('@api_get_dataset', context={'dataset-id': dataset_id}).json()

If you can't find a reference to an endpoint in the Kive API wrapper itself, I think we can remove it from the map.

[this]: https://github.com/cfe-lab/Kive/blob/master/api/kiveapi/kiveapi.py#L288

**Comment by nathanielknight** [comment]
Yes, it's the URLs--the values in the dict that seem to be missing.

Specifically, I think the following endpoints are outdated:

- `/api/compounddatatypes/`
- `/api/pipelines/`
- `api/pipelinefamilies`

**Comment by donkirkby** [comment]
Also outdated:

* `/api/runs/`
* `/api/runbatches/`
